archetype/engine-v2/src/main/java/io/helidon/build/archetype/engine/v2/ArchetypeEngineV2.java

```diff
@@ -91,6 +98,37 @@ public Path generate(InputResolver inputResolver,
         context.requireRootScope();
         onResolved.run();
 
+        //prepare user inputs for .helidon
+        Set<ContextValue.ValueKind> valueKindFilter = Set.of(ContextValue.ValueKind.EXTERNAL, ContextValue.ValueKind.USER);
```

Comment:
Most of the logic here should go in `ContextSerializer.serialize()` ; and maybe have a method `Context.asMap()` to flatten a context as a map.